lesson13/script.js

```diff
@@ -54,26 +54,27 @@ async function init(){
     }
 }
 
-function modalWindow(){
-    const modal = document.getElementById('js-modal');
-    const openButton = document.getElementById('js-open-btn');
-    const closeButton = document.getElementById('js-close-btn');
-    const requestButton = document.getElementById('js-request-btn');
-    
-    openButton.addEventListener('click',function(){
-        modal.classList.add('is-show');
-        openButton.classList.add('is-deactive');
+function openModalWindow(){   
+    document.getElementById('js-open-btn').addEventListener('click',function(){
+        document.getElementById('js-modal').classList.add('is-show');
+        this.classList.add('is-deactive');
     },false);
+}
 
-    closeButton.addEventListener('click',function(){
-        modal.classList.remove('is-show');
-        openButton.classList.remove('is-deactive');
+function closeModalWindow(){ 
+    document.getElementById('js-close-btn').addEventListener('click',function(){
+        document.getElementById('js-modal').classList.remove('is-show');
+        document.getElementById('js-open-btn').classList.remove('is-deactive');
     },false);
+}
 
-    requestButton.addEventListener('click', function(){
-        modal.classList.remove('is-show');
+function requestData(){ 
+    document.getElementById('js-request-btn').addEventListener('click', function(){
+        document.getElementById('js-modal').classList.remove('is-show');
         init();
     },false);
 }
 
-modalWindow();
+openModalWindow();
+closeModalWindow();
+requestData();
```

Comment:
I changed the modal function to a process-by-process method.
Because the processing content and the function name do not match.
By cutting out for each process, the function name and the process content are matched.

Reply:
I too thought it was a good idea to separate the functions as it is difficult to bundle them in one function.

In such cases, there is a way to use an immediate function.
I am sharing an article that was shared with me by a student.
[JavaScriptで即時関数を使う理由](https://qiita.com/katsukii/items/cfe9fd968ba0db603b1e)
[JavaScript best practices(Avoid globals)](https://www.w3.org/wiki/JavaScript_best_practices#:~:text=a%20good%20thing.-,Avoid%20globals,-Global%20variables%20and)

```
(function(){
    const modal = document.getElementById('js-modal');
    const openButton = document.getElementById('js-open-btn');
    const closeButton = document.getElementById('js-close-btn');
    const requestButton = document.getElementById('js-request-btn');

    function openModalWindow() {
      openButton.addEventListener("click",function () {
          modal.classList.add("is-show");
          this.classList.add("is-deactive");
        },false);
    }

    function closeModalWindow() {
      closeButton.addEventListener("click", function () {
          modal.classList.remove("is-show");
          openButton.classList.remove("is-deactive");
        },false);
    }

    function requestData() {
      requestButton.addEventListener("click",function () {
          modal.classList.remove("is-show");
          init();
        }, false );
    }

    openModalWindow();
    closeModalWindow();
    requestData();

}());
```

Reply:
Thank you for telling me a very good way!
I changed to use an immediate function.
https://github.com/nakagawamai/js-lesson/pull/15/commits/36ba8aa98255d1f27aa9319e5ca3e5be2b68464c